core/client/fs/src/main/java/alluxio/client/block/policy/CapacityBaseRandomPolicy.java

```diff
@@ -85,45 +68,25 @@ public Optional<WorkerNetAddress> getWorker(GetWorkerOptions options) {
     if (totalCapacity.get() == 0L) {
       return Optional.empty();
     }
-    long randomLong = randomInCapacity(totalCapacity.get());
+    long randomLong = randomInCapacity(options.getBlockInfo().getBlockId(), totalCapacity.get());
     WorkerNetAddress targetWorker = rangeStartMap.floorEntry(randomLong).getValue().getNetAddress();
-    addWorkerToCache(options.getBlockInfo().getBlockId(), targetWorker);
     return Optional.of(targetWorker);
   }
 
-  protected long randomInCapacity(long totalCapacity) {
-    return ThreadLocalRandom.current().nextLong(totalCapacity);
-  }
-
-  protected WorkerNetAddress findCacheWorker(GetWorkerOptions options) {
-    List<WorkerNetAddress> cacheCandidateList =
-        mBlockLocationCache.getIfPresent(options.getBlockInfo().getBlockId());
-    if (cacheCandidateList != null && mMaxReplicaSize > 0) {
-      Set<WorkerNetAddress> eligibleAddresses = new HashSet<>();
-      for (BlockWorkerInfo info : options.getBlockWorkerInfos()) {
-        eligibleAddresses.add(info.getNetAddress());
-      }
-      List<WorkerNetAddress> eligibleCacheList =
-          cacheCandidateList.stream().filter(eligibleAddresses::contains)
-              .collect(Collectors.toList());
-      if (eligibleCacheList.size() >= mMaxReplicaSize) {
-        int index = ThreadLocalRandom.current().nextInt(eligibleCacheList.size());
-        return eligibleCacheList.get(index);
-      }
+  protected long randomInCapacity(Long blockId, long totalCapacity) {
+    if (mMaxReplicaSize < 0) {
+      return ThreadLocalRandom.current().nextLong(totalCapacity);
     }
-    return null;
+    // blockId base hash value to decide which worker to cache data,
+    // so the same block will be routed to the same set of worker.
+    long sourceValue = blockId + ThreadLocalRandom.current().nextInt(mMaxReplicaSize);
```

Comment:
I wonder why this policy wants to take block replication into consideration. I think master has a dedicated background job taking care of the replication adjustment periodically. On the other hand, the other policies like `LocalFirstPolicy` etc. do not care about replications, either. WDYT @jiacheliu3 @beinan?

Reply:
@dbw9580 You are right, master can take care of replications, so we can remove this random logic. But different from LocalFirstPolicy， this policy is much like DeterministicHashPolicy which use 'alluxio.user.ufs.block.read.location.policy.deterministic.hash.shards' to random target worker. 

Anyway, I agree with you to remove this random logic in this policy.